lib/graphql/schema/rescue_middleware.rb

```diff
@@ -0,0 +1,53 @@
+module GraphQL
+  class Schema
+    # - Store a table of errors & handlers
+    # - Rescue errors in a middleware chain, then check for a handler
+    # - If a handler is found, use it & return a {GraphQL::ExecutionError}
+    # - If no handler is found, re-raise the error
+    class RescueMiddleware
+      # @return [Hash] `{class => proc}` pairs for handling errors
+      attr_reader :rescue_table
+      def initialize
+        @rescue_table = {}
```

Comment:
Using a table means that you can't rescue derived classes.  E.g. `MySchema.rescue_from(StandardError) { "internal error" }` would only catch StandardError, not derived classes like RuntimeError.